python/src/main/java/org/apache/apex/malhar/python/base/BasePythonExecutionOperator.java

```diff
@@ -0,0 +1,591 @@
+      emitStragglers();
+      try {
+        PythonRequestResponse result = processPython(tuple,getApexPythonEngine());
+        if ( result != null) {
+          responseCounterForThisWindow += 1;
+          outputPort.emit(result);
+        } else {
+          numNullResponsesPerWindow += 1;
+        }
+      } catch (ApexPythonInterpreterException e) {
+        responseCounterForThisWindow += 1; // Error tuples need to be accounted for totals
+        errorPort.emit(tuple);
+        LOG.error("Error while processing tuple", e);
+      }
+    }
+  };
+
+
+  @Override
+  public void activate(Context.OperatorContext context)
+  {
+    getApexPythonEngine().setNumStarvedReturns(0L);
+  }
+
+  @Override
+  public void deactivate()
+  {
+
+  }
+
+  /***
+   * Used to emit the stragglers into the Stragglers port. INvoked both when a new tuple arrives or when Idle time
+   *  is detected on this operator.
+   */
+  private void emitStragglers()
+  {
+    List<PythonRequestResponse> stragglerResponse = new ArrayList<>();
+    getApexPythonEngine().getDelayedResponseQueue().drainTo(stragglerResponse);
+    for (PythonRequestResponse aReqResponse : stragglerResponse) {
+      responseCounterForThisWindow += 1;
+      stragglersPort.emit(aReqResponse);
+    }
+  }
+
+  /***
+   * Instantiates the configured python engine. Only in-memory implementation provided for now
+   * @param context The operator context
+   * @return The python engine
+   */
+  protected ApexPythonEngine initApexPythonEngineImpl(Context.OperatorContext context)
+  {
+    JepPythonEngine jepPythonEngine = new JepPythonEngine("" + context.getId(),workerThreadPoolSize);
+    jepPythonEngine.setSleepTimeAfterInterpreterStart(sleepTimeDuringInterpreterBoot);
+    return jepPythonEngine;
+  }
+
+  /***
+   * Instantiates the partitioner. Only Thread Starvation based partitioner for now.
+   */
+  private void initPartitioner()
+  {
+    if (partitioner == null) {
+      synchronized (objectForLocking) {
+        if (partitioner == null) {
+          switch (partitionerType) {
+            default:
+            case THREAD_STARVATION_BASED:
+              partitioner = new ThreadStarvationBasedPartitioner(this);
+              break;
+          }
+        }
+      }
+    }
+  }
+
+  /***
+   * Starts the python engine and also sleeps for sometime ( configurable) to ensure that the interpreter is
+   *  completely booted up in memory. The time taken to boot the interpreter depends on the libaries that are loaded etc
+   * @param context The Operator context
+   */
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    super.setup(context);
+    apexPythonEngine = initApexPythonEngineImpl(context);
+    Map<PythonInterpreterConfig,Object> preInitConfigurations = getPreInitConfigurations();
+    apexPythonEngine.preInitInterpreter(preInitConfigurations);
+    apexPythonEngine.startInterpreter();
+    apexPythonEngine.postStartInterpreter();
+    initPartitioner();
```

Comment:
this shouldn't be here, since the partitioner is executed in the master and not in the worker container (setup is only called in the latter).

Reply:
Removed the initPartitioner from this. Thanks for pointing this out.